tests/unit/http_crawler/test_http_crawler.py

```diff
@@ -201,3 +207,37 @@ async def test_http_status_statistics(crawler: HttpCrawler, server: respx.MockRo
     assert len(server['html_endpoint'].calls) == 10
     assert len(server['404_endpoint'].calls) == 10
     assert len(server['500_endpoint'].calls) == 30
+
+
+@pytest.mark.parametrize(
+    'http_client_class',
+    [CurlImpersonateHttpClient, HttpxHttpClient],
+    ids=['curl', 'httpx'],
+)
+async def test_filling_web_form(http_client_class: type[BaseHttpClient]) -> None:
+    http_client = http_client_class()
+    crawler = HttpCrawler(http_client=http_client)
+
+    form_data = {
+        'custname': 'John Doe',
+        'custtel': '1234567890',
+        'custemail': 'johndoe@example.com',
+        'size': 'large',
+        'topping': '["bacon", "cheese", "mushroom"]',
+        'delivery': '13:00',
+        'comments': 'Please ring the doorbell upon arrival.',
+    }
+
+    @crawler.router.default_handler
+    async def request_handler(context: HttpCrawlingContext) -> None:
+        response = json.loads(context.http_response.read())
+        # The httpbin.org/post endpoint returns the form data in the response.
+        assert response['form'] == form_data
```

Comment:
Could you also assert that the request_handler was called?

Reply:
Oh, good catch.